src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -181,7 +181,8 @@ def __init__(self, *args, **kwargs):
     def setUp(self):
         """Set up run before each test."""
         super(TestWithoutServers, self).setUp()
-        load_mpi('openmpi')
+        if not load_mpi("openmpi"):
+            self.fail("Failed to load openmpi")
```

Comment:
this will break A21 soak testing because there is no openmpi module on Aurora.   I will open a ticket to address the soak issue.

Reply:
Yeah, so this was just fixing a bug where the MPI load failure was silently being ignored.

Probably/perhaps this needs to have a bigger picture analysis to decide what the test framework is supposed to do about not having openmpi.  Maybe trying to load it here is wrong.  Maybe it should be site-configurable as to which implementation to use.  Maybe the code here should try multiple different MPI implementations until it finds one, and only fail if none are available.  But one should be able to run non-MPI tests still in that case, IMHO.

So yeah, a more wholesome analysis of what we really want needs to be done and changes made accordingly.  But as you suggest, that is unrelated to this effort.